onward/app/feed/MostPopularAgent.scala

```diff
@@ -3,34 +3,33 @@ package feed
 import contentapi.ContentApiClient
 import common._
 import services.OphanApi
-import play.api.libs.json.{JsArray, JsValue}
 import model.RelatedContentItem
-import scala.concurrent.Future
-import scala.util.control.NonFatal
+import scala.concurrent.{ExecutionContext, Future}
 
-class MostPopularAgent(contentApiClient: ContentApiClient) extends Logging with ExecutionContexts {
+class MostPopularAgent(contentApiClient: ContentApiClient) extends Logging {
 
   private val agent = AkkaAgent[Map[String, Seq[RelatedContentItem]]](Map.empty)
 
   def mostPopular(edition: Edition): Seq[RelatedContentItem] = agent().getOrElse(edition.id, Nil)
 
-  def refresh() {
+  def refresh()(implicit ec: ExecutionContext): Future[List[Map[String, Seq[RelatedContentItem]]]] = {
```

Comment:
I have refactored this file to collapse the `List` of `Map` into a single one 868a157
Thanks for the feedback @LATaylor-guardian